pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -251,7 +272,169 @@ def measure_final_state(self, circuit: QuantumScript) -> Result:
             Tuple[TensorLike]: The measurement results
         """
 
+        if not circuit.shots:
+            # analytic case
+            if len(circuit.measurements) == 1:
+                return self.measurement(circuit.measurements[0])
+
+            return tuple(self.measurement(mp) for mp in circuit.measurements)
```

Comment:
Isn't PL added support for PyTrees as return values? I'm concerned for users requested a non-tuple return value.

Reply:
Not sure, but I think the device's job is to bring the measurements in a format that the QNode can handle, and the QNode does the final reshaping, casting, etc.